app/db/models/user.js

```diff
@@ -50,6 +54,38 @@ module.exports = (sequelize, DataTypes) => {
     }
   )
 
+  /*
+  example naive usage: user.addRole('ADMIN')
+
+  different users of streetmix can have different roles,
+  which can enable or disable certain features
+
+  We need to be able to:
+  * add roles
+  * remove roles
+  * potentially have a timestamp of when a role was last updated (?)
+  * check if a role _should_ be updated(added/removed) or not based on criteria(this might need to be its own method, that gets a list of JSON objects to test against).
+    this should be a function that accepts some parameters and
+    searches for a match in the user's model based on a linking field.
+      * if a linking attribute is found(returns a boolean), that give role can be updated (calls the 'add roles' method). returns an error if false (cant find a user) or breaks (can't perform the function for whatever else reason)
+      * if it is not found, the role should be removed(calls the 'remove roles' method) or user should be flagged for a semi-manual check
+        ( potentially this would be provisional, subject to manual approval?
+          just in case we accidentally remove functionality from people.
+          this might need a nice failsafe,
+          especially early on when our supporter list is small enough)
+*/
+
+  User.prototype.addRole = function (newRole) {
```

Comment:
plenty of ways this could be expanded, type checking, passing values directly from the roles object, etc

Reply:
most of this, i think, _can_ be covered by stuff built in to sequelize

Reply:
@louh Ideally I'd clean this up as part of the PR, but we could merge as is given that this has been open for a bit. I'm still not %100 sure where I'd want things to end up but it'll also be easier to determine that once we're using the roles functionality somewhere.

Reply:
jk all 'done'